gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -316,6 +289,17 @@ private FileSystem getFileSystem() throws IOException {
     }
   }
 
+  private DatasetsFinder getDatasetsFinder() {
+    try {
+      return (DatasetsFinder) Class.forName(this.state.getProp(COMPACTION_DATASETS_FINDER, DEFAULT_COMPACTION_DATASETS_FINDER))
+        .getConstructor(State.class, FileSystem.class, String.class, String.class, String.class, List.class, List.class,
```

Comment:
There are too many constructor parameters. You can just pass `state` and get everything else from it in the dataset finder.